src/Query/Builder.php

```diff
@@ -132,12 +132,16 @@ protected function prepareInsertForDml($values)
      */
     protected function runSelect()
     {
+        $sql = $this->toSql();
+        $bindings = $this->getBindings();
+        $options = [];
+
         if ($this->timestampBound !== null) {
-            return $this->connection->selectWithTimestampBound(
-                $this->toSql(), $this->getBindings(), $this->timestampBound
-            );
+            $options += $this->timestampBound->transactionOptions();
         }
 
-        return parent::runSelect();
+        return count($options) > 0
+            ? $this->connection->selectWithOptions($sql, $bindings, $options)
+            : $this->connection->select($sql, $bindings);
```

Comment:
```suggestion
        return $this->connection->selectWithOptions($sql, $bindings, $options); 
```
I think it would be simpler to just `selectWithOptions`.